File: backend/src/main/java/br/com/calciolari/datahub/imports/infrastructure/persistence/ValidationResultEntity.java
package br.com.calciolari.datahub.imports.infrastructure.persistence;

import java.math.BigDecimal;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "validation_result")
public class ValidationResultEntity {

	@Id
	private UUID id;

	@Column(name = "parse_attempt_id", nullable = false)
	private UUID parseAttemptId;

	@Column(nullable = false)
	private String code;

	@Column(nullable = false)
	private String status;

	@Column(name = "source_value")
	private BigDecimal sourceValue;

	@Column(name = "calculated_value")
	private BigDecimal calculatedValue;

	private BigDecimal difference;

	private BigDecimal tolerance;

	@Column(name = "rule_version", nullable = false)
	private String ruleVersion;

	@Column(name = "source_locator")
	private String sourceLocator;

	protected ValidationResultEntity() {
	}

	public ValidationResultEntity(
			UUID id,
			UUID parseAttemptId,
			String code,
			String status,
			BigDecimal sourceValue,
			BigDecimal calculatedValue,
			BigDecimal difference,
			BigDecimal tolerance,
			String ruleVersion,
			String sourceLocator) {
		this.id = id;
		this.parseAttemptId = parseAttemptId;
		this.code = code;
		this.status = status;
		this.sourceValue = sourceValue;
		this.calculatedValue = calculatedValue;
		this.difference = difference;
		this.tolerance = tolerance;
		this.ruleVersion = ruleVersion;
		this.sourceLocator = sourceLocator;
	}

	public UUID getId() {
		return id;
	}

	public UUID getParseAttemptId() {
		return parseAttemptId;
	}

	public String getCode() {
		return code;
	}

	public String getStatus() {
		return status;
	}
}

File: backend/src/Calciolari.DataHub/Persistence/Entities/ValidationResultEntity.cs
namespace Calciolari.DataHub.Persistence.Entities;

public sealed class ValidationResultEntity
{
    public Guid Id { get; set; }
    public Guid ParseAttemptId { get; set; }
    public string Code { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public decimal? SourceValue { get; set; }
    public decimal? CalculatedValue { get; set; }
    public decimal? Difference { get; set; }
    public decimal? Tolerance { get; set; }
    public string RuleVersion { get; set; } = string.Empty;
    public string? SourceLocator { get; set; }

    public ValidationResultEntity()
    {
    }

    public ValidationResultEntity(
        Guid id,
        Guid parseAttemptId,
        string code,
        string status,
        decimal? sourceValue,
        decimal? calculatedValue,
        decimal? difference,
        decimal? tolerance,
        string ruleVersion,
        string? sourceLocator)
    {
        Id = id;
        ParseAttemptId = parseAttemptId;
        Code = code;
        Status = status;
        SourceValue = sourceValue;
        CalculatedValue = calculatedValue;
        Difference = difference;
        Tolerance = tolerance;
        RuleVersion = ruleVersion;
        SourceLocator = sourceLocator;
    }
}
